### DISCUSSION TITLE
WIPs

### DISCUSSION THREAD

**Discussion by dixyes** [dixyes/lwmbs#2] [Ideas]
- [x] libc provided iconv and fully remove libiconv
- [ ] documents for usage
- [ ] packing up into single binary
- [x] gd
- [x] gd extra libs
- [ ] libpq
- [ ] mongo
- [ ] imagemagick
- [ ] libstdcxx
- [ ] swoole
- [ ] libev
- [x] libxml

**Comment by emanuele-scarsella** [comment]
Hi, I've seen that `intl` is listed as one of the extensions in [src.json](https://github.com/dixyes/lwmbs/blob/master/src.json#L383) but it has `icu` as a not-optional dependency and `icu` is not one of the implemented libraries.
Do you have any plan to implement it in the future?
Can I get any guidance implementing it myself?
Thanks in advance.

**Reply by dixyes** [reply]
Sorry for late reply, I'm too busy these days.
I have listed all in-tree(i.e. which included in PHP source) extensions in that metadata file, but intl is few used by us, so it's lower priority for me.
I had a look on your fork, unix parts seems to be OK, but for windows, cygwin build may not work, you need a MSVC build.
All contribution is welcomed, if making a MSVC build is really hard (thanks MS), you may check how [PHP official](https://github.com/winlibs) done, or just leave it to me (but it maybe takes a few weeks)
BTW, please change the copyright holder to your name, that's your work

**Reply by emanuele-scarsella** [reply]
Thank you for the reply.
For the time being I'm going to use [swoole/swoole-cli](https://github.com/swoole/swoole-cli) since it seams a valid alternative for my project, it is not configurable but it has all the extensions I need, But I may come back to attempt implementing it again in the future.
### What I tried so far:
To make the linux and macos versions I took some information from [static-php-cli](https://github.com/crazywhalecc/static-php-cli) since they have a very similar system and `intl` is implemented in there, but still both linux and macos don't seem to work for now ([they fail while compiling micro-php](url)) so some more debugging is needed there.
For windows I knew I was screwd in the moment I read in the [icu windows build documentation](https://unicode-org.github.io/icu/userguide/icu4c/build#how-to-build-and-install-on-windows):
> Building International Components for Unicode requires:
> -    Microsoft Windows 7 or newer. (Windows XP and Windows Vista are not supported)
> -    Microsoft Visual C++ (part of [Visual Studio](https://www.visualstudio.com/)) (from either Visual Studio 2017 or Visual Studio 2019)

But I still made a desperate try in the imaginative hope that somehow `windows-latest` had all the requirements already installed and as predictable, I didn't worked.

Following your suggestion I've tried avoiding cygwin and building with the next available option (MSYS) adding in the github action using [msys2/setup-msys2@v2](https://github.com/msys2/setup-msys2), [ilammy/msvc-dev-cmd@v1](https://github.com/ilammy/msvc-dev-cmd) and [egor-tensin/setup-cygwin](https://github.com/egor-tensin/setup-cygwin) but it [still fails](https://github.com/emanuele-scarsella/lwmbs/actions/runs/7048140819/job/19183625879#step:10:155) to find `link.exe`, probably some more configuration is needed there as well, like adding it in the PATH env variable for example.

### To conclude
As I've said I'm simply going to use [swoole-cli](https://github.com/swoole/swoole-cli) for my project since I have a deadline to meet, but I hope to come back in the free time and maybe get something to work, but I've very limited skills in this field so I'm not very confident in being able to do any good here.  ¯\\_ (ツ)_/¯

Thank you again for your answer.
